app/forms/statuses/form.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Statuses
+  class Form < ApplicationForm
+    attr_reader :submit_label
+
+    def initialize(submit_label: nil)
+      super()
+
+      @submit_label = submit_label || I18n.t(:button_save)
+    end
+
+    form do |statuses_form|
+      statuses_form.text_field(
+        label: attribute_name(:name),
+        name: :name,
+        required: true,
+        input_width: :medium
+      )
+
+      statuses_form.text_field(
+        label: attribute_name(:default_done_ratio),
+        name: :default_done_ratio,
```

Comment:
💡 In this form, the labels could have been deduced from the name. Don't know if that is always the case but it looks like a good default in case we want to optimize this. 

Not necessary to address in this PR.

Reply:
Yes, indeed, but I'm wary about deviating from the simplicity of using the existing helpers. There is a bit of duplication, but it's not a big deal as it reads well. I may change my mind later.